### PR TITLE
add upgrade notes

### PR DESCRIPTION
**Pull request by vizvamitra**

### CODE REVIEW

File: UPGRADE_NOTES.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+list.pages # => 15
+list.page # => 5
+list.total # => 308 (files in a project)
+```
+
+It **won't work anymore**. For the details on the new file/group lists interface see [readme][readme]
+
+The core features of the new file/group list API are: 
+
+- transparent pagination via..
```
Comment (dmitry-mukhin) [inline UPGRADE_NOTES.md]: via ...?

File: UPGRADE_NOTES.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- enumerable interface
+- loading objects at demand
```
Comment (dmitry-mukhin) [inline UPGRADE_NOTES.md]: on demand

File: UPGRADE_NOTES.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- ordering, filtering and sliceing
```
Comment (dmitry-mukhin) [inline UPGRADE_NOTES.md]: slicing

File: UPGRADE_NOTES.md (hunk 1 above)
Comment (vizvamitra) [reply UPGRADE_NOTES.md]: transparent pagination via enumerable interface

**Pull request merged**